File: 012.jarakirefactor/funcao22.erl
-module(funcao22).

-compile(export_all).

-import(loop, [for/3, while/2]).

bo(Var_a1) ->
    st:insert({"b", false}),
    case st:get("a") == true of
      true -> st:insert({"b", false});
      false -> st:insert({"b", true})
    end,
    st:get("b").

quadrado(Var_a1) ->
    st:insert({"temp", st:get("a") * st:get("a")}),
    st:get("temp").

cubo(Var_a1) ->
    st:insert({"a",
	       st:get("a") * (st:get("a") * st:get("a"))}),
    io:format("~s", ["O valor de a ao cubo: "]),
    io:format("~p~n", [st:get("a")]).

main(Var_Args1) ->
    st:new(),
    st:insert({"x", 5}),
    st:insert({"t", quadrado(st:get("x"))}),
    cubo(st:get("x")),
    io:format("~s", ["O valor de a ao quadrado eh: "]),
    io:format("~p~n", [st:get("t")]),
    st:insert({"b", bo(true)}),
    io:format("~s", ["O valor de b ao quadrado eh: "]),
    io:format("~p~n", [st:get("b")]).

File: old_vsn/010.funcao/funcao2.2.java
package funcao22;

public class Funcao22 {

		public static int quadrado(int a) {
			int temp;

			temp = a*a;

			return temp;
		}

		public static void cubo(int a) {
			a = a*a*a;
			System.out.print("O valor de a ao cubo: ");
			System.out.println(a);
		}

		public static void main(String[] Args) {
			int x;
			int t;

			x = 5;
			t = quadrado(x);
			cubo(x);
			System.out.print("O valor de a ao quadrado eh: ");
			System.out.println(t);
		}
}
